packages/agent/src/connect-v2.ts

```diff
@@ -0,0 +1,195 @@
+import { utils } from '@web5/crypto';
+import { DwnProtocolDefinition, DwnRecordsPermissionScope } from './index.js';
+import {
+  Web5ConnectAuthResponse,
+  Oidc,
+  type PushedAuthResponse,
+} from './oidc-v2.js';
+import { pollWithTtl } from './utils.js';
+import { Convert } from '@web5/common';
+import { DidDht } from '@web5/dids';
+
+async function initClient({
+  connectServerUrl,
+  walletUri,
+  permissionRequests,
+  onWalletUriReady,
+  validatePin,
+}: WalletConnectOptions) {
+  // ephemeral client did for ECDH, signing, verification
+  // TODO: use separate keys for ECDH vs. sign/verify. could maybe use secp256k1.
+  const clientDid = await DidDht.create();
+
+  // Hash the code verifier to use as a code challenge and to encrypt the Request Object.
+  const { codeVerifierBytes } = Oidc.generateRandomCodeVerifier();
+
+  // Derive the code challenge based on the code verifier
+  const { codeChallengeBytes, codeChallengeBase64Url } =
+      await Oidc.deriveCodeChallenge(codeVerifierBytes);
```

Comment:
The concern I have here is not necessarily with us encrypting the request object using a symmetric key then passing the key to the wallet so it can decrypt it (though I am curious if this pattern is something we invented in web5, or a known OIDC pattern. The comment here from code review stand point is that the code makes it unnecessarily confusing for the reader to conclude that's what the code is doing. 

The code seems to conflate with the intent of `code_challenge` and `code_verifier`. The `code_verifier` is not persisted anywhere for later verification, so in the end all the fancy names and extra code just serve the purpose of creating a 32 byte random number no?

If that's the case, why don't we just do `const symmetricKey = utils.randomBytes(32);` and call it a day?

Reply:
fixed

Reply:
I see that you've move the verifier code into `generateCodeChallenge()`, which helps encapsulate the logic for better readability I guess, but my main question remains: the fact that the `code_verifier` is not persisted anywhere tells me something is half baked or doesn't add up here.

I'd suggest that we just do `const symmetricKey = utils.randomBytes(32);` and close out this comment and worry about implementing `code_challenge` and `code_verifier` properly in a subsequent PR.

But not okay for having a bunch of fancy properties/params for appears to add extra security or feature when it in the end is just used as a symmetric key.

Let's jump on a call if I am still missing the point and close out this comment.

Reply:
It's a good call out but the suggestion I think misses the mark. The intent of the code_verifier / code_challenge is to meet the `PKCE` requirements. Rather than hiding that by calling it `symmetricKey` and therefore abandoning PKCE because the implementation is different or incomplete the better solution is to see if we can implement PKCE.

Reply:
Here is what we would need: https://datatracker.ietf.org/doc/html/rfc7636#section-4.1

Reply:
I have added a TODO comment on this topic. hopefully we can close this out. 

1. it was never intended to **only** be a symmetricKey
2. we should see if we can implement PKCE

Reply:
I'd agree to implementing PKCE (`code_challenge/verifier`) in a latter PR and would be down to take the work item too , however (sorry for beating the dead horse), why are we using the `code_challenge` as the symmetric encryption key? I thought the point of encrypting the request is so that the relay server (as well as unintended actor) is not able to decrypt the content? The `code_challenge` if I understand correctly, is meant to be visible to the relay server, thus using `code_challenge` as the encryption key would allow relay server to decrypt the content if/when PKCE is implemented. This is why I kept suggesting leaving `code_challenge/verifier` alone and just call the random bytes what it is: a symmetric key. Apologies if I am still not making sense, please educate me if I continue to misunderstand things.

Reply:
ok that makes sense. so just leave the verifier unused?

Reply:
My suggestion is still calling it a key:

`const symmetricKey = utils.randomBytes(32);`

And call it a `key` or similar in wallet URI.

> (though I am curious if this pattern is something we invented in web5, or a known OIDC pattern.)

I am a noob to OIDC too, so I don't know if there are better "more compliant or established" ways to ship an encryption key hence my original comment above, but what I know is shipping/calling it as `code_challenge` is definitely not it.